autoload/cmake/fileapi.vim

```diff
@@ -54,6 +54,9 @@ endfunction
 function! s:ParseIndex(build_dir) abort
     let l:index_path = s:FindIndexFile(a:build_dir)
     if l:index_path ==# ''
+        " TODO: only show this with an existing build tree without api files
+        call s:logger.EchoError('Api response missing, run :CMakeGenerate')
+        call s:logger.LogError('Api response missing, run :CMakeGenerate')
```

Comment:
The error message is in my opinion too abstract for the user, most people who will get this error won't know what "Api response missing" means. Could we make this message a bit more meaningful to the user? Is this too vague? "List of build targets missing, run `:CMakeGenerate`"

Reply:
I know this is not a good message. I just noticed this would be a problem and added something to not forget. I think 'List of build targets' is too vague. I, unfortunately, don't have a better idea myself. The only thing I can think of would be to explicitly name the `cmake-file-api(7)` here too. This is only slightly better, but at least you could google (or `:Man`) what this is.

Reply:
I would not get too technical, as all the user cares about is in fact the latter part of the sentence. I also don't have any great suggestions now, so I'll leave this up to you, and we can refine in the future if necessary

Reply:
I like to think that people using vim to write software in a language that uses CMake for a buildsystem are at least somewhat interested in technicalities. (Could be wrong though)
As I think 'Api response' is in fact not useful at all I will change this to include the reference to the file api. If we ever find some better message we can change it. I will also explicitly mention that target completion will not work because of this.